apps/src/calc/expressionNode.js

```diff
@@ -163,42 +174,51 @@ ExpressionNode.prototype.evaluate = function (globalMapping, localMapping) {
     if (left.err) {
       throw left.err;
     }
-    left = left.result;
+    left = left.result.toExact();
 
     if (this.children_.length === 1) {
       switch (this.value_) {
         case 'sqrt':
-          return { result: Math.sqrt(left) };
+          val = jsnums.sqrt(left);
+          break;
         case 'sqr':
-          return { result: left * left };
+          val = jsnums.sqr(left);
+          break;
         default:
           throw new Error('Unknown operator: ' + this.value_);
-        }
+      }
+      return { result: ensureJsnum(val) };
```

Comment:
Maybe a comment here reiterating what you said in the PR description in case some foolhardy future coder decides to be a hero and remove this redundant-seeming wrapping :)